staging/src/k8s.io/apiserver/pkg/storage/etcd3/event.go

```diff
@@ -42,7 +43,11 @@ func parseKV(kv *mvccpb.KeyValue) *event {
 	}
 }
 
-func parseEvent(e *clientv3.Event) *event {
+func parseEvent(e *clientv3.Event) (*event, error) {
+	if !e.IsCreate() && e.PrevKv == nil {
+		// If the previous value is nil, error. One example of how this is possible is if the previous value has been compacted already.
+		return nil, fmt.Errorf("etcd event was not a create and has a nil prevKV, key: %s, modRevision: %d, isDelete %v", string(e.Kv.Key), e.Kv.ModRevision, e.Type == clientv3.EventTypeDelete)
```

Comment:
@jpbetz can you sanity check that all non-create events should have a PrevKv set?

Reply:
(we only use this method in one place, on a watch, started with the PrevKv option)
```
opts := []clientv3.OpOption{clientv3.WithRev(wc.initialRev + 1), clientv3.WithPrevKV()}
```

Reply:
Looks like providing a `prevKV` was intentional. See my comment here: https://github.com/kubernetes/kubernetes/issues/76624#issuecomment-486939316

I'm betting this really is due to compaction. It looks like we're using `clientv3.WithPrevKV` correctly.